### PR TITLE
Extract Date Input Change Listener and Validation Logic into Hooks

### PR DESCRIPTION
**Pull request by tn3rb**
## Problem this Pull Request solves
Consolidates existing logic for updating and validating entity date properties into custom hooks that can be shared by multiple components

## How has this been tested
monkey based browser testing only

### CODE REVIEW

File: assets/src/hooks/use-end-date-change-listener.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+			touchField( startDateFormKey );
+			touchField( endDateFormKey );
+		}
+	}, [
+		entity[ dateProps.start ],
+		entity[ dateProps.end ],
+		startDateFormKey,
+		endDateFormKey,
+		updateField,
+		touchField,
```
Comment (ManzoorWaniEE) [inline assets/src/hooks/use-end-date-change-listener.js]: Why should it depend upon `updateField` and `touchField`?

File: assets/src/hooks/use-start-date-change-listener.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+			touchField( startDateFormKey );
+			touchField( endDateFormKey );
+		}
+	}, [
+		entity[ dateProps.start ],
+		entity[ dateProps.end ],
+		startDateFormKey,
+		endDateFormKey,
+		updateField,
+		touchField,
```
Comment (ManzoorWaniEE) [inline assets/src/hooks/use-start-date-change-listener.js]: Same here...

File: assets/src/editor/events/dates-and-times/editor-date/edit-form/use-date-entity-input-config.js [hunk 3]
```diff
@@ -64,45 +105,8 @@ const useDateEntityInputConfig = ( {
...
-					if ( startDate > dateEntity.end ) {
-						return __(
-							'End Date & Time must be set later than the Start Date & Time',
-							'event_espresso'
-						);
-					}
-				}
-			},
+			changeListener: startDateChangeListener,
+			validate: startDateChangeValidator,
```
Comment (ManzoorWaniEE) [inline assets/src/editor/events/dates-and-times/editor-date/edit-form/use-date-entity-input-config.js]: Nice one! It makes the config file clean and simple.

File: assets/src/hooks/use-end-date-change-listener.js (hunk 1 above)
Comment (tn3rb) [reply assets/src/hooks/use-end-date-change-listener.js]: from the React documentation regarding dependencies:

> The array of dependencies is not passed as arguments to the function. Conceptually, though, that’s what they represent: every value referenced inside the function should also appear in the dependencies array.

My understanding of that is that unless you are 1000% certain that some external code used by a hook is completely static and will not change, then it should be included as a dependency (just to be clear I was not using the term "static" here to mean "a class function not belonging to an instance", but just to mean something that stays the same).

I don't know if `updateField` and `touchField` are completely unchanging and therefore I have included them as dependencies. It's probably fair to assume that those callbacks won't change as they are likely just simple setters... but I'm just not 1000% certain of that...

Comment (ManzoorWaniEE) [reply assets/src/hooks/use-end-date-change-listener.js]: 👍🏻

**Pull request merged**